src/Microsoft.AspNet.Server.Kestrel/ServiceContext.cs

```diff
@@ -29,7 +32,20 @@ public ServiceContext(ServiceContext context)
 
         public IApplicationLifetime AppLifetime { get; set; }
 
-        public IKestrelTrace Log { get; set; }
+        public IKestrelTrace Log
+        {
+            get
+            {
+                return _log;
+            }
+            set
+            {
+                _log = value;
+                ThreadPoolActions = new ThreadPoolActions(_log);
```

Comment:
Nit: Can we just set this in KestrelServer rather than having a custom getter and setter for Log?